### ISSUE TITLE
Keep paths loaded in memory for performance (e.g. in tab bar)

### ISSUE THREAD

**Issue by jeanbaptistebeau** [frzi/SwiftUIRouter#62]
Is there a way to keep some paths loaded in memory, i.e. not recreate the views when the path changes?

I was implementing a tab bar layout like in the [RandomUsers example](https://github.com/frzi/SwiftUIRouter-Examples/blob/main/RandomUsers/Shared/RootView.swift), but with this solution the pages get recreated every time the user switches tab, which can makes the process of switching tab a bit laggy.

Is there a way to keep those pages in memory to provide a smooth tab bar experience?

**Comment by frzi** [comment]
No, this is beyond the scope of SwiftUI Router and would probably require a significant rewrite of the framework. It also raises questions like what the state is based on. The path of the Navigator or the path of the Route (which may contain placeholders)?

Sadly as the framework currently works you're somewhat forced to keep track of states yourself (which I understand is very difficult when it comes to scrollviews et al) 😞

**Comment by stevenb9** [comment]
I was able to workaround this by using Factory (https://github.com/hmlongco/Factory). 

1. Register your view as a singleton (static let myService = Factory(scope: .singleton) { Service() as Service }
2. Inject the singleton view in your TabContents
3. For the route, hse this injected variable as the content (Route("/tabbar/map/*", content: map))

This will keep the view in memory
